business_requirement_deliverable_default/models/business_requirement_deliverable_default.py

```diff
@@ -0,0 +1,47 @@
+# -*- coding: utf-8 -*-
+# © 2016 Elico Corp (https://www.elico-corp.com).
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openerp import api, fields, models
+
+
+class BusinessRequirementDeliverable(models.Model):
+    _inherit = "business.requirement.deliverable"
+
+    def _prepare_resource_lines(self):
+        rl_data = self.product_id.sudo().resource_lines.copy_data()
+        rl_data = [(0, 0, item) for index, item in enumerate(rl_data)]
+        return rl_data
+
+    @api.multi
+    @api.onchange('product_id')
+    def product_id_change(self):
+        super(BusinessRequirementDeliverable, self).product_id_change()
+        product = self.product_id
```

Comment:
AFAIK and onchange is triggered from a form, and operates on a singleton, so a for loop shouldn't be needed.